### PR TITLE
Header permalink

### PR DESCRIPTION
**Pull request by iftodimihail**

### CODE REVIEW

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/244#" title="Author self-approved">iftodimihail</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: packages/shared/src/mdx/components/index.js [hunk 1]
```diff
@@ -80,11 +100,29 @@ function generateHeadingId(children) {
...
+        <h1 id={generateHeadingId(props.children)} {...props} />
+      </HeaderWrap>
+    );
+  },
+  h2: props => {
+    const [showPermalink, setShowPermalink] = useState(false);
+    return (
+      <HeaderWrap
+        onMouseEnter={() => setShowPermalink(true)}
+        onMouseLeave={() => setShowPermalink(false)}
```
Comment (codrin-iftimie) [inline packages/shared/src/mdx/components/index.js:119]: Why not use CSS hover?

**Review by codrin-iftimie — CHANGES_REQUESTED** [review]
generateHeadingId would not return the correct url if the header is inside a tab...
You'll see that tabs now have an identifier, which means that you can change the tab from the URL using ?<identifier>=<tab_name>. The heading url should append to this string.

Make a context provider in the tabs component so you can use it in the headings to detect the identifier and tabname

File: packages/shared/src/components/ui/Tabs.js [hunk 2]
```diff
@@ -28,24 +28,30 @@ const StyledTabs = styled(AntTabs)`
...
+  const [activeKey, setActiveKey] = React.useState(rest?.children?.[0]?.key || "");
 
+  React.useEffect(() => {
     if (query[identifier]) {
-      defaultTab = query[identifier];
+      setActiveKey(query[identifier])
+      clickedOnHash.current = false;
     }
-    setActiveKey(defaultTab)
+  }, [location.hash]);
```
Comment (codrin-iftimie) [inline packages/shared/src/components/ui/Tabs.js:45]: []

File: packages/shared/src/components/ui/Tabs.js [hunk 3]
```diff
@@ -28,24 +28,30 @@ const StyledTabs = styled(AntTabs)`
...
-      const anchor = document.createElement("a")
+  React.useEffect(() => {
+    if (location.hash && clickedOnHash.current === false) {
+      const anchor = document.createElement("a");
       anchor.href = location.hash;
       anchor.click();
+      clickedOnHash.current = true;
     }
-  }, [])
+  }, [location.hash]);
```
Comment (codrin-iftimie) [inline packages/shared/src/components/ui/Tabs.js:54]: [activeKey]

**Review by codrin-iftimie — CHANGES_REQUESTED** [review]

**Review by codrin-iftimie — APPROVED** [review]

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/244#pullrequestreview-526165677" title="Approved">codrin-iftimie</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/244#" title="Author self-approved">iftodimihail</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: packages/shared/src/mdx/components/index.js [hunk 4]
```diff
@@ -75,16 +115,34 @@ function generateHeadingId(children) {
...
+    const tabsIdentifierData = useTabsContext();
+
+    return (
+      <HeaderWrap>
+        <a href={generatePermalinkAnchor(props.children, tabsIdentifierData)}>
+          <FontAwesomeIcon icon="link" />
+        </a>
+        <h2 id={generateHeadingId(props.children)} {...props} />
+      </HeaderWrap>
+    )
```
Comment (codrin-iftimie) [inline packages/shared/src/mdx/components/index.js:144]: DRY. You can use a HOC system to prevent code duplication.

Comment (codrin-iftimie) [reply packages/shared/src/mdx/components/index.js:144]: Merging as is for now

**Review by codrin-iftimie — APPROVED** [review]

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/244#pullrequestreview-526166247" title="Approved">codrin-iftimie</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/244#" title="Author self-approved">iftodimihail</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**